documents/description.md

```diff
@@ -1,9 +1,83 @@
 # Project description
 
 ## Motivation
+DOTA 2 is a very hard, complex game. In order to even be competent, one must memorize all 120+ heroes (characters) in the game, each of which has at least 4 unique abilities (skills), as well as all of the 100+ items that all do unique things. Then, in order to be good at the game, you must know which of these hero/skill/item (which is commonly referred to as a "build") combinations are effective, and against what other hero/skill/item combinations. Some of this can be figured out through logic and intuition from playing the game for a long time, but for the most part, there is no concrete way to determine how much damage one build does against another; the mental math required is too complex. Even professional players experiment with new builds, and might not be using the optimal ones in every situation.
+
+Thus, I want to create an easy way for people who play the game, primarily newer players, but not exclusively, to test various dota 2 builds against similar builds, and other players' builds. Because the goal is to enable lots of comparisons at once, I feel that a language is best to do this, as all of the existing tools force the user to input an entire build one at a time, with very little ability to copy-past builds with slight modifications.
 
 ## Language domain
+The domain of this language is DOTA 2. The user should be able to (in an ideal world) write a program to test any combination of dota 2 heroes, items and skills with as much specificity as possible. It should be impossible to do anything other than test DOTA 2 elements. 
 
 ## Language design
+The language should be designed with three main elements. First, the user should be able to specify a build. This includes:
+ + The hero
+ + The hero's level
+ + The Hero's items
+ + The levels of each ability the hero has
+
+The user can then store builds like variables in a conventional programming language. 
+
+The second element of the language is essentially dota-specific queries. So, one can take one of the builds that is specified, and ask for various stats on that build (for example, damage, hit points, attack speed, DPS, etc). 
+
+Lastly, the language needs control-flow elements. These are essentially ways to compare different builds. Some things that the language might have is a foreach-style loop and list elements, so one can make a list of builds and run the same command on all of them. Additionally, I'd want things that compare two builds. Commands like max(stat, build1,build2), and WhoWins(build1, build2). Obviously, this isn't the syntax, but it describes the general structure of the control flow. 
```

Comment:
Not only does this seem useful to players but also useful to the designers who are considering adding new heroes.